[PATCH v3] example: sysinfo: use new sysinfo APIs

psavol opened the pull request:
Use the new odp_sys_config_print() and odp_system_info()
calls.

JereLeppanen commented:
To avoid every name appearing two times in a kind of error-prone fashion, could use something like

`#define CASE_CPU_ARCH(NAME) case ODP_CPU_ARCH_ ## NAME: return #NAME`

Either way though, the patch looks OK to me.

psavol commented:
v2: rebased and updated implementation. Now also CPU HW ISA version is printed. On ARMv8 devices ISA version is filled in if part number is recognized. Otherwise HW ISA version is unknown.

psavol commented:
> To avoid every name appearing two times in a kind of error-prone fashion, could use something like
> 
> `#define CASE_CPU_ARCH(NAME) case ODP_CPU_ARCH_ ## NAME: return #NAME`

I changed to print commonly used CPU arch and ISA names instead of enum names: e.g. ARMv8.1 instead of ARMV8_1.
